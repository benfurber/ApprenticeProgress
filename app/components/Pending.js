// @flow

import React, { Component } from "react";
import { StyleSheet, Text } from "react-native";
import { H2, View } from "native-base";

import { Background, Dot } from "components";
import { colours, standards } from "styles";

type Props = {
  condition: string,
};
class Pending extends Component<Props> {
  render() {
    const { condition } = this.props;

    return (
      <Background noRalph={true} style={styles.background}>
        {this._renderDots(condition)}
        {this._renderConditionText(condition)}
        <View style={styles.dotsView} />
      </Background>
    );
  }

  _renderDots(condition: string) {
    if (condition !== "Loading") {
      return <View style={styles.dotsView} />;
    }

    return (
      <View style={styles.dotsView}>
        <Dot delay={0} />
        <Dot delay={500} />
        <Dot delay={900} />
        <Dot delay={300} />
        <Dot delay={1200} />
      </View>
    );
  }

  _renderConditionText(condition: string) {
    if (condition === "Error") {
      return (
        <View style={styles.textView}>
          <H2 style={styles.redText}>{condition}</H2>
        </View>
      );
    }

    return (
      <View style={styles.textView}>
        <H2>{condition}</H2>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  background: {
    flexDirection: "column",
    alignItems: "center",
  },
  dotsView: {
    alignItems: "flex-end",
    flex: 4,
    flexDirection: "row",
    justifyContent: "center",
  },
  redText: {
    color: colours.red,
  },
  textView: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "center",
    ...standards.elementPadding,
  },
});

export { Pending };
