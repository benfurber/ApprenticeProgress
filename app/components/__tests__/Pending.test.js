// @flow

import React from "react";
import { shallow } from "enzyme";

import { Pending } from "../Pending";

describe("Loading", () => {
  it("has a valid snapshot when for loading", () => {
    const component = shallow(<Pending condition="Loading" />);

    expect(component).toMatchSnapshot();
  });

  it("has a valid snapshot when for an Error", () => {
    const component = shallow(<Pending condition="Error" />);

    expect(component).toMatchSnapshot();
  });
});
