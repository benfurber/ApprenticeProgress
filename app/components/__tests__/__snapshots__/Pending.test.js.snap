// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading has a valid snapshot when for an Error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pending
    condition="Error"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(ViewNB)
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 4,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        />,
        <Styled(ViewNB)
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 30,
              "paddingVertical": 10,
            }
          }
        >
          <Styled(H2)
            style={
              Object {
                "color": "#ED3E44",
              }
            }
          >
            Error
          </Styled(H2)>
        </Styled(ViewNB)>,
        <Styled(ViewNB)
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 4,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        />,
      ],
      "noRalph": true,
      "style": Object {
        "alignItems": "center",
        "flexDirection": "column",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "alignItems": "flex-end",
            "flex": 4,
            "flexDirection": "row",
            "justifyContent": "center",
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(H2)
            style={
              Object {
                "color": "#ED3E44",
              }
            }
          >
            Error
          </Styled(H2)>,
          "style": Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 30,
            "paddingVertical": 10,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Error",
            "style": Object {
              "color": "#ED3E44",
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": "Error",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "alignItems": "flex-end",
            "flex": 4,
            "flexDirection": "row",
            "justifyContent": "center",
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(ViewNB)
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 4,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          />,
          <Styled(ViewNB)
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingHorizontal": 30,
                "paddingVertical": 10,
              }
            }
          >
            <Styled(H2)
              style={
                Object {
                  "color": "#ED3E44",
                }
              }
            >
              Error
            </Styled(H2)>
          </Styled(ViewNB)>,
          <Styled(ViewNB)
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 4,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          />,
        ],
        "noRalph": true,
        "style": Object {
          "alignItems": "center",
          "flexDirection": "column",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "alignItems": "flex-end",
              "flex": 4,
              "flexDirection": "row",
              "justifyContent": "center",
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(H2)
              style={
                Object {
                  "color": "#ED3E44",
                }
              }
            >
              Error
            </Styled(H2)>,
            "style": Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 30,
              "paddingVertical": 10,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Error",
              "style": Object {
                "color": "#ED3E44",
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "Error",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "alignItems": "flex-end",
              "flex": 4,
              "flexDirection": "row",
              "justifyContent": "center",
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Loading has a valid snapshot when for loading 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pending
    condition="Loading"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(ViewNB)
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 4,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Dot
            delay={0}
          />
          <Dot
            delay={500}
          />
          <Dot
            delay={900}
          />
          <Dot
            delay={300}
          />
          <Dot
            delay={1200}
          />
        </Styled(ViewNB)>,
        <Styled(ViewNB)
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 30,
              "paddingVertical": 10,
            }
          }
        >
          <Styled(H2)>
            Loading
          </Styled(H2)>
        </Styled(ViewNB)>,
        <Styled(ViewNB)
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 4,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        />,
      ],
      "noRalph": true,
      "style": Object {
        "alignItems": "center",
        "flexDirection": "column",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Dot
              delay={0}
            />,
            <Dot
              delay={500}
            />,
            <Dot
              delay={900}
            />,
            <Dot
              delay={300}
            />,
            <Dot
              delay={1200}
            />,
          ],
          "style": Object {
            "alignItems": "flex-end",
            "flex": 4,
            "flexDirection": "row",
            "justifyContent": "center",
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "delay": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "delay": 500,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "delay": 900,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "delay": 300,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "delay": 1200,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(H2)>
            Loading
          </Styled(H2)>,
          "style": Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 30,
            "paddingVertical": 10,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Loading",
            "virtual": undefined,
          },
          "ref": null,
          "rendered": "Loading",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "alignItems": "flex-end",
            "flex": 4,
            "flexDirection": "row",
            "justifyContent": "center",
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(ViewNB)
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 4,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Dot
              delay={0}
            />
            <Dot
              delay={500}
            />
            <Dot
              delay={900}
            />
            <Dot
              delay={300}
            />
            <Dot
              delay={1200}
            />
          </Styled(ViewNB)>,
          <Styled(ViewNB)
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingHorizontal": 30,
                "paddingVertical": 10,
              }
            }
          >
            <Styled(H2)>
              Loading
            </Styled(H2)>
          </Styled(ViewNB)>,
          <Styled(ViewNB)
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 4,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          />,
        ],
        "noRalph": true,
        "style": Object {
          "alignItems": "center",
          "flexDirection": "column",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Dot
                delay={0}
              />,
              <Dot
                delay={500}
              />,
              <Dot
                delay={900}
              />,
              <Dot
                delay={300}
              />,
              <Dot
                delay={1200}
              />,
            ],
            "style": Object {
              "alignItems": "flex-end",
              "flex": 4,
              "flexDirection": "row",
              "justifyContent": "center",
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "delay": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "delay": 500,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "delay": 900,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "delay": 300,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "delay": 1200,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(H2)>
              Loading
            </Styled(H2)>,
            "style": Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 30,
              "paddingVertical": 10,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Loading",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "Loading",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "alignItems": "flex-end",
              "flex": 4,
              "flexDirection": "row",
              "justifyContent": "center",
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
